pyatv/core/scan.py

```diff
@@ -288,3 +311,171 @@ def _end_if_identifier_found(self, response: mdns.Response):
+        infos = [
+            AsyncServiceInfo(zc_type, cast(DNSPointer, record).alias)
+            for zc_type in zc_types
+            for record in self.zeroconf.cache.async_all_by_details(
+                zc_type, _TYPE_PTR, _CLASS_IN
+            )
+        ]
+        await asyncio.gather(
+            *[info.async_request(self.zeroconf, zc_timeout) for info in infos]
+        )
+        services_by_address: Dict[str, List[AsyncServiceInfo]] = {}
+        for info in infos:
+            address = _first_non_link_local_or_v6_address(info.addresses)
+            if address:
+                services_by_address.setdefault(address, []).append(info)
+        return services_by_address
+
+    async def _async_models_by_name(
+        self, names: Iterable[str], zc_timeout: float
+    ) -> Dict[str, str]:
+        """Probe the DEVICE_INFO_TYPE."""
+        name_to_model: Dict[str, str] = {}
+        device_infos = {
+            name: AsyncDeviceInfoServiceInfo(
+                DEVICE_INFO_TYPE, f"{name}.{DEVICE_INFO_TYPE}"
+            )
+            for name in names
+        }
+        await asyncio.gather(
+            *[
+                info.async_request(
+                    self.zeroconf, zc_timeout, question_type=DNSQuestionType.QU
+                )
+                for info in device_infos.values()
+            ]
+        )
+        for name, info in device_infos.items():
+            possible_model = info.properties.get(b"model")
+            if possible_model:
+                with contextlib.suppress(UnicodeDecodeError):
+                    name_to_model[name] = possible_model.decode("utf-8")
+        return name_to_model
+
+    def _async_process_responses(
+        self,
+        atv_services_by_address: Dict[str, List[mdns.Service]],
+        name_to_model: Dict[str, str],
+        name_by_address: Dict[str, str],
+    ):
+        """Process and callback each aggregated response to the base handler."""
+        for address, atv_services in atv_services_by_address.items():
+            model = None
+            name_for_address = name_by_address.get(address)
+            if name_for_address is not None:
+                possible_model = name_to_model.get(name_for_address)
+                if possible_model:
+                    model = possible_model
+            self.handle_response(
+                mdns.Response(
+                    services=atv_services,
+                    deep_sleep=all(
+                        service.port == 0 and service.type != SLEEP_PROXY_TYPE
+                        for service in atv_services
+                    ),
+                    model=model,
+                )
+            )
+
+    async def process(self, timeout: int) -> None:
+        """Start to process devices and services."""
+        zc_timeout = timeout * 1000
+        services_by_address = await self._async_services_by_addresses(zc_timeout)
+        atv_services_by_address: Dict[str, List[mdns.Service]] = {}
```

Comment:
```suggestion
        dev_services_by_address: Dict[str, List[mdns.Service]] = {}
```
Or something similar. As pyatv supports more devices than Apple TVs nowadays, we should refrain from using terms like "atv". There are remaining occurrences and I try to remove them along the way.

Reply:
44b128fa5ee0418ec8ed40e15f5e4a7d23c9a105